### PR TITLE
[13.0][ADD] product_dimension_volumetric_weight: Add Product Uom Volumetric Weight

### PR DESCRIPTION
**Pull request by victoralmau**
This module extends the functionality of product_dimension to add Volumetric weight in products.

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT30722

### CODE REVIEW

**Comment by pedrobaeza** [comment]
If this finally depends on `product_dimension`, shouldn't it be `product_dimension_volumetric_weight`?

**Review by chienandalu — COMMENTED** [review]
¿No support for variants?

**Comment by hparfr** [comment]
Hi,

May be a dumb question but, why there is no weight in the formula of volumetric_weight ? 

About the UI, the unit of "volumetric_weight" is not explicit. Shoulnd't it be "dimensional_uom / Kg" ?

**Comment by victoralmau** [comment]
Changes done.
IMO volumetric weight is need to be added in extra addon because `product_dimension` is only related to dimension and field name is correct.

**Review by chienandalu — APPROVED** [review]

File: product_dimension_volumetric_weight/models/product.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    weight_volumetric = fields.Float(
+        string="Volumetric weight",
+        compute="_compute_weight_volumetric",
+        store=True,
+        readonly=True,
+    )
+
+    @api.depends("dimensional_uom_id", "volume")
+    def _compute_weight_volumetric(self):
```
Comment (pedrobaeza) [inline product_dimension_volumetric_weight/models/product.py]: No need to repeat this method if it's the same as upstream `product.template` one.

File: product_dimension_volumetric_weight/readme/DESCRIPTION [hunk 2]
```diff
@@ -0,0 +1 @@
+This module extends the functionality of product_dimension to add Volumetric weight in products.
```
Comment (pedrobaeza) [inline product_dimension_volumetric_weight/readme/DESCRIPTION]:
```suggestion
This module extends the functionality of product_dimension to add Volumetric weight in products.

It's used for example by some carrier agencies to compute the cost price of the delivery service.
```

File: product_dimension_volumetric_weight/tests/test_product_uom.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        self.uom_m.write({"weight_volumetric_ratio": 400})
+        self.uom_cm = self.env.ref("uom.product_uom_cm")
+        self.uom_cm.write({"weight_volumetric_ratio": 400})
+
+    def test_volumetric_weight_in_m(self):
+        product_form = Form(self.env["product.template"])
+        product_form.name = "Test product"
+        product_form.product_length = 1
+        product_form.product_height = 1
+        product_form.product_width = 1
```
Comment (pedrobaeza) [inline product_dimension_volumetric_weight/tests/test_product_uom.py]: This is testing a functionality of other module, so no need to include it here.

File: product_dimension_volumetric_weight/tests/test_product_uom.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        product_form.dimensional_uom_id = self.uom_m
+        self.assertAlmostEqual(400, product_form.weight_volumetric)
+
+    def test_volumetric_weight_in_cm(self):
+        product_form = Form(self.env["product.template"])
+        product_form.name = "Test product"
+        product_form.product_length = 40
+        product_form.product_height = 50
+        product_form.product_width = 60
```
Comment (pedrobaeza) [inline product_dimension_volumetric_weight/tests/test_product_uom.py]: Again testing something that is not of this module. Why don't you set `volume` and just that?

File: product_dimension_volumetric_weight/models/product.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        for item in self:
+            item.weight_volumetric = (
+                item.volume * item.dimensional_uom_id.weight_volumetric_ratio
+            )
+
+
+class ProductProduct(models.Model):
```
Comment (pedrobaeza) [inline product_dimension_volumetric_weight/models/product.py]: Split both models in their corresponding files.

File: product_dimension_volumetric_weight/models/product.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    weight_volumetric = fields.Float(
+        string="Volumetric weight",
+        compute="_compute_weight_volumetric",
+        store=True,
+        readonly=True,
+        digits="Volume",
```
Comment (pedrobaeza) [inline product_dimension_volumetric_weight/models/product.py]: This is weight, not volume.

File: product_dimension_volumetric_weight/models/product.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    weight_volumetric = fields.Float(
+        string="Volumetric weight",
+        compute="_compute_weight_volumetric",
+        store=True,
+        readonly=True,
```
Comment (pedrobaeza) [inline product_dimension_volumetric_weight/models/product.py]: A compute is always readonly

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Add also configure file (and generate README), stating how to set the ratio at UoM.

**Comment by pedrobaeza** [comment]
Another thing: why fields are called `weight_volumetric`, but the module is `volumetric_weight`. The only reasoning I see is for having together `weight` and `weight_volumetric`, but not sure if that reason is important enough for having this duality in the name. What others think about this?

**Comment by victoralmau** [comment]
> Another thing: why fields are called `weight_volumetric`, but the module is `volumetric_weight`. The only reasoning I see is for having together `weight` and `weight_volumetric`, but not sure if that reason is important enough for having this duality in the name. What others think about this?

Yes, probably `volumetric_weight` is the better option to field names.

**Review by rousseldenis — APPROVED** [review]
Code review

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**